src/format.rs

```diff
@@ -208,13 +208,26 @@ pub(crate) struct FormatProcessData<'a> {
     pub(crate) with_target: bool,
     #[cfg(feature = "ansi")]
     pub(crate) ansi: bool,
+    pub(crate) with_short_file_names: bool,
 }
 
 impl<'a> fmt::Display for FormatProcessData<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         let thread_name = self.thread_name;
         let target = self.metadata.target();
-        let file = self.metadata.file().unwrap_or("");
+        let file = self
+            .metadata
+            .file()
+            .map(|f| {
+                if self.with_short_file_names {
+                    f.rsplit_once('/')
+                        .map(|(_, file_name)| file_name)
+                        .unwrap_or_default()
+                } else {
+                    f
+                }
```

Comment:
I'm not opposed to this change, but I'd prefer an approach like this instead:

```suggestion
            .map(|f| {
                if self.with_short_file_names {
                    let f = Path::new(f);
                    f.file_name().map(OsStr::to_str);
                } else {
                    f
                }
```

I haven't compiled this, so it might fail, but I'd rather be operating in terms of `Path`s instead of string manipulation.